odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+
+    def det_refpoint(self, angle):
+        """The detector reference point function.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's motion parameter set
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (`ndim`,)
+            The reference point on the circle with radius ``R`` at a given
+            rotation angle :math:`\\phi`, defined as :math:`R(-\\sin\\phi,
+            \\cos\\phi, 0)`
+        """
+        if angle not in self.motion_params:
+            raise ValueError('angle {} not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+
+        angle += self.angle_offset
+
+        # ASTRA 'cone_vec' convention
+        return self.det_radius * np.array([-sin(angle), cos(angle), 0])
+
+    def src_position(self, angle):
+        """The source position function.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's motion parameter set
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (`ndim`,)
+
+            The source position at ``z`` on the circle with radius ``r`` at
+            a given rotation angle :math:`\\phi`, defined as :math:`(r
+            \\sin\\phi, -r\\cos\\phi, 0)`
+        """
+        if angle not in self.motion_params:
+            raise ValueError('angle {} not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+
+        angle += self.angle_offset
+
+        # ASTRA cone_vec convention
+        return self.src_radius * np.array([sin(angle), -cos(angle), 0])
+
+        # TODO: backprojection weighting function?
+
+
+class HelicalConeFlatGeometry(ConeFlatGeometry):
+    """Helical cone beam geometry in 3d with flat detector.
+
+    The source moves on a circle with radius ``r``, and the detector
+    reference point is opposite to the source on a circle with radius ``R``
+    and aligned tangential to the circle.
+
+    The motion parameter is the (1d) rotation angle parametrizing
+    source and detector positions.
+    """
+
+    def __init__(self, angle_intvl, dparams, src_rad, det_rad,
+                 spiral_pitch_factor, agrid=None, dgrid=None,
+                 angle_offset=0, axis=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        angle_intvl : `Interval` or 1-dim. `IntervalProd`
+            The motion parameters
+        dparams : `Rectangle` or 2-dim. `IntervalProd`
+            The detector parameters
+        src_rad : `float`
+            Radius of the source circle, must be positive
+        det_rad : `float`
+            Radius of the detector circle, must be positive
+        spiral_pitch_factor : `float`
+            Dimensionless factor given by the table feed per rotation
```

Comment:
> What do you @adler-j use Elekta?

We dont have CT's

> But we need to abstract this into our own (idealized, mathematical) representation, and conversions need to be handled by data readers. Other data formats may use different conventions, and we need to settle for one internal representation. And in my opinion, this internal representation should be reduced to the minimum complexity, which is describing a spiral by radius and pitch.

I agree,  (dz / d theta) is the most simple way to represent this.